util/src/main/java/tech/pegasys/artemis/util/SSZTypes/SSZBackingVector.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.SSZTypes;
+
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.util.backing.VectorViewWrite;
+import tech.pegasys.artemis.util.backing.ViewRead;
+
+public class SSZBackingVector<C, R extends ViewRead> extends SSZAbstractCollection<C>
```

Comment:
Would be nice to try to add more descriptive type parameters.  For example:
```suggestion
public class SSZBackingVector<TValue, TView extends ViewRead> extends SSZAbstractCollection<TValue>
```

Reply:
Agree. But let me consider this for subsequent refactor since these classes are most likely to be removed